File: solutions/go/triangle/1/triangle.go
package triangle

type Kind int

const (
	NaT Kind = iota // not a triangle
	Equ // equilateral
	Iso // isosceles
	Sca // scalene
)

func KindFromSides(a, b, c float64) Kind {
	switch {
	case a == b && b == c && a > 0:
		return Equ
	case a >= b + c || b >= a + c || c >= a + b:
		return NaT
	case a != b && b != c && c != a:
		return Sca
	default:
		return Iso
	}
}

File: solutions/python/triangle/1/triangle.py
def equilateral(sides):
    sides.sort(reverse=True)
    return len(set(sides)) <= 1 and sides[0] < sum(sides[1:])


def isosceles(sides):
    sides.sort(reverse=True)
    return len(set(sides)) <= 2 and sides[0] < sum(sides[1:])


def scalene(sides):
    sides.sort(reverse=True)
    return len(set(sides)) == 3 and sides[0] < sum(sides[1:])
